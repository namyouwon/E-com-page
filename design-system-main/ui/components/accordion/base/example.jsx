// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import uniqueId from 'lodash.uniqueid';
import Accordion, { AccordionSection } from '../';
import { ActionOverflow } from '../../../components/menus/action-overflow/example';
import Card from '../../cards/';

export const AccordionExample = () => (
  <Accordion>
    <AccordionSection
      summary="Accordion summary"
      isOpen
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - A
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - B
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - C
    </AccordionSection>
  </Accordion>
);

export default <AccordionExample />;

export let states = [
  {
    id: 'section-one-open',
    label: 'Section one open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-two-open',
    label: 'Section two open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-three-open',
    label: 'Section three open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'multiple-sections-open',
    label: 'Multiple sections open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];

export let examples = [
  {
    id: 'long-heading-truncated',
    label: 'Long Section Heading with Truncation for larger screens',
    demoStyles: 'max-width: 500px;',
    storybookStyles: true,
    element: (
      <Card>
        <Accordion>
          <AccordionSection
            summary="Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </Card>
    )
  },
  {
    id: 'nested-accordions',
    label: 'Nested Accordions',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          <Accordion>
            <AccordionSection
              summary="Nested accordion summary"
              isOpen
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - A
            </AccordionSection>
            <AccordionSection
              summary="Nested accordion summary"
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - B
            </AccordionSection>
          </Accordion>
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'styled',
    label: 'Wrapped in Card',
    element: (
      <div className="slds-card">
        <Accordion>
          <AccordionSection
            summary="Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </div>
    )
  },
  {
    id: 'has-action-overflow',
    label: 'Has Action Overflow Menu',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Background Color',
    demoStyles: '--slds-c-accordion-summary-color-background: lightblue',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Text Color',
    demoStyles:
      '--slds-c-accordion-heading-text-color: green; --slds-c-accordion-heading-text-color-hover: red',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];
