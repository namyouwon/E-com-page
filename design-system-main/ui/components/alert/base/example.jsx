// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import Alert from '../';
import { UtilityIcon } from '../../icons/base/example';
import EmptyLink from '../../../shared/empty-link/';
import Heading from '../../../shared/heading/';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Alert>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];

export let states = [
  {
    id: 'theme-info',
    label: 'Default (Theme Class  - Deprecated)',
    element: (
      <Alert isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'warning',
    label: 'Warning',
    element: (
      <Alert type="warning">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-warning',
    label: 'Warning (Theme Class  - Deprecated)',
    element: (
      <Alert type="warning" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'error',
    label: 'Error',
    element: (
      <Alert type="error">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-error',
    label: 'Error (Theme Class  - Deprecated)',
    element: (
      <Alert type="error" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'offline',
    label: 'Offline',
    element: (
      <Alert type="offline">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-offline',
    label: 'Offline (Theme Class  - Deprecated)',
    element: (
      <Alert type="offline" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];
