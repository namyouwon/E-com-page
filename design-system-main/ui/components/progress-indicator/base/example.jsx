// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import classNames from 'classnames';
import ButtonIcon from '../../button-icons/';
import { Tooltip } from '../../tooltips/base/example';
import {
  Modal,
  ModalHeader,
  ModalContent,
  ModalFooter,
  Backdrop
} from '../../modals/base/example';
import { ProgressBar } from '../../progress-bar';

/// ///////////////////////////////////////////
// Partial(s)
/// ///////////////////////////////////////////

export let Progress = props => (
  <div className={classNames('slds-progress', props.className)}>
    <ol className="slds-progress__list">{props.children}</ol>
    <ProgressBar className="slds-progress-bar_x-small" value={props.value} />
  </div>
);

export let Step = props => (
  <li
    className={classNames(
      'slds-progress__item',
      props.className,
      props.active ? 'slds-is-active' : null,
      props.done ? 'slds-is-completed' : null,
      props.error ? 'slds-has-error' : null
    )}
  >
    {props.done && !props.error ? (
      <ButtonIcon
        className="slds-progress__marker slds-progress__marker_icon"
        symbol="success"
        aria-describedby={props['aria-describedby']}
        assistiveText={props.done ? props.children + ' - Completed' : null}
        title={props.done ? props.children + ' - Completed' : null}
      />
    ) : props.error ? (
      <ButtonIcon
        className="slds-progress__marker slds-progress__marker_icon"
        symbol="error"
        aria-describedby={props['aria-describedby']}
        assistiveText={props.error ? props.children + ' - Error' : null}
        title={props.error ? props.children + ' - Error' : null}
      />
    ) : (
      <button
        className="slds-button slds-progress__marker"
        aria-describedby={props['aria-describedby']}
      >
        <span className="slds-assistive-text">
          {props.children} {props.active ? '- Active' : null}
        </span>
      </button>
    )}
  </li>
);

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <div className="demo-only" style={{ padding: '1rem' }}>
        <Progress value="0">
          <Step active>Step 1</Step>
          <Step>Step 2</Step>
          <Step>Step 3</Step>
          <Step>Step 4</Step>
          <Step>Step 5</Step>
        </Progress>
      </div>
    )
  }
];

export let states = [
  {
    id: 'next-step',
    label: 'Next Step',
    element: (
      <div className="demo-only" style={{ padding: '1rem' }}>
        <Progress value="25">
          <Step done>Step 1</Step>
          <Step active>Step 2</Step>
          <Step>Step 3</Step>
          <Step>Step 4</Step>
          <Step>Step 5</Step>
        </Progress>
      </div>
    )
  },
  {
    id: 'has-error',
    label: 'Step - Error',
    element: (
      <div className="demo-only" style={{ padding: '1rem' }}>
        <Progress value="25">
          <Step done>Step 1</Step>
          <Step error>Step 2</Step>
          <Step>Step 3</Step>
          <Step>Step 4</Step>
          <Step>Step 5</Step>
        </Progress>
      </div>
    )
  },
  {
    id: 'tooltip',
    label: 'Tooltip',
    element: (
      <div className="demo-only" style={{ padding: '3rem 1rem 0' }}>
        <Progress value="50">
          <Step done>Step 1</Step>
          <Step done>Step 2</Step>
          <Step active aria-describedby="step-3-tooltip">
            Step 3
          </Step>
          <Step>Step 4</Step>
          <Step>Step 5</Step>
        </Progress>
        <Tooltip
          className="slds-nubbin_bottom"
          id="step-3-tooltip"
          style={{
            position: 'absolute',
            top: '0px',
            left: 'calc(50% + 7px)',
            transform: 'translateX(-50%)'
          }}
        >
          Verify Email
        </Tooltip>
      </div>
    )
  }
];

export let examples = [
  {
    id: 'modal',
    label: 'In a modal',
    element: (
      <div className="demo-only" style={{ height: '640px' }}>
        <Backdrop>
          <Modal className="slds-modal_large" aria-labelledby="header43">
            <ModalHeader>
              <h1 id="header43" className="slds-text-heading_medium">
                Modal header
              </h1>
            </ModalHeader>
            <ModalContent className="slds-grow slds-p-around_medium" />
            <ModalFooter className="slds-grid slds-grid_align-spread">
              <button className="slds-button slds-button_neutral" aria-label="Cancel and close">Cancel</button>
              <Progress className="slds-progress_shade" value="25">
                <Step done>Step 1</Step>
                <Step active>Step 2</Step>
                <Step>Step 3</Step>
                <Step>Step 4</Step>
                <Step>Step 5</Step>
              </Progress>
              <button className="slds-button slds-button_brand">Save</button>
            </ModalFooter>
          </Modal>
        </Backdrop>
      </div>
    )
  },
  {
    id: 'shade',
    label: 'On a gray background',
    element: (
      <div
        className="demo-only"
        style={{ background: '#F3F2F2', padding: '1rem' }}
      >
        <Progress className="slds-progress_shade" value="25">
          <Step done>Step 1</Step>
          <Step active>Step 2</Step>
          <Step>Step 3</Step>
          <Step>Step 4</Step>
          <Step>Step 5</Step>
        </Progress>
      </div>
    )
  }
];
