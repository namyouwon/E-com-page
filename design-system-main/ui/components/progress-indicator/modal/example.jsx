// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Progress, Step } from '../base/example';
import {
  Modal,
  ModalHeader,
  ModalContent,
  ModalFooter,
  Backdrop
} from '../../modals/base/example';

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <div className="demo-only" style={{ height: '640px' }}>
        <Backdrop>
          <Modal className="slds-modal_large" aria-labelledby="header43">
            <ModalHeader>
              <h1 id="header43" className="slds-text-heading_medium">
                Modal header
              </h1>
            </ModalHeader>
            <ModalContent className="slds-grow slds-p-around_medium" />
            <ModalFooter className="slds-grid slds-grid_align-spread">
              <button className="slds-button slds-button_neutral" aria-label="Cancel and close">Cancel</button>
              <Progress className="slds-progress_shade" value="25">
                <Step done>Step 1</Step>
                <Step active>Step 2</Step>
                <Step>Step 3</Step>
                <Step>Step 4</Step>
                <Step>Step 5</Step>
              </Progress>
              <button className="slds-button slds-button_brand">Save</button>
            </ModalFooter>
          </Modal>
        </Backdrop>
      </div>
    )
  }
];
