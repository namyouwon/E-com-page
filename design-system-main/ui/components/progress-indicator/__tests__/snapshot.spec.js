// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import { Progress as BaseProgress, Step as BaseStep } from '../base/example';
import {
  Progress as VerticalProgress,
  Step as VerticalStep
} from '../vertical/example';

const { matchesMarkup } = createHelpers(__dirname);

describe('Base Progress Indicator', () => {
  it('renders a base step', () => matchesMarkup(<BaseStep>Step 1</BaseStep>));

  it('renders an active base step', () =>
    matchesMarkup(<BaseStep active>Step 1</BaseStep>));

  it('renders a done base step', () =>
    matchesMarkup(<BaseStep done>Step 1</BaseStep>));

  it('renders an error base step', () =>
    matchesMarkup(<BaseStep error>Step 1</BaseStep>));

  it('renders an active base step with aria-describedby', () =>
    matchesMarkup(
      <BaseStep active aria-describedby="step-1-tooltip">
        Step 1
      </BaseStep>
    ));

  it('renders a base progress indicator', () =>
    matchesMarkup(
      <BaseProgress>
        <BaseStep active>Step 1</BaseStep>
        <BaseStep>Step 2</BaseStep>
      </BaseProgress>
    ));
});

describe('Vertical Progress Indicator', () => {
  it('renders a vertical step', () =>
    matchesMarkup(<VerticalStep>Step 1</VerticalStep>));

  it('renders an active vertical step', () =>
    matchesMarkup(<VerticalStep active>Step 1</VerticalStep>));

  it('renders a done vertical step', () =>
    matchesMarkup(<VerticalStep done>Step 1</VerticalStep>));

  it('renders an error vertical step', () =>
    matchesMarkup(<VerticalStep error>Step 1</VerticalStep>));

  it('renders a vertical progress indicator', () =>
    matchesMarkup(
      <VerticalProgress>
        <VerticalStep active>Step 1</VerticalStep>
        <VerticalStep>Step 2</VerticalStep>
      </VerticalProgress>
    ));

  it('renders a vertical progress indicator with some progress', () =>
    matchesMarkup(
      <VerticalProgress value="50">
        <VerticalStep done>Step 1</VerticalStep>
        <VerticalStep active>Step 2</VerticalStep>
        <VerticalStep>Step 2</VerticalStep>
      </VerticalProgress>
    ));

  it('renders a vertical progress indicator with borders', () =>
    matchesMarkup(
      <VerticalProgress value="50" hasBorders>
        <VerticalStep done>Step 1</VerticalStep>
        <VerticalStep active>Step 2</VerticalStep>
        <VerticalStep>Step 2</VerticalStep>
      </VerticalProgress>
    ));

  it('renders a vertical progress indicator with green bar', () =>
    matchesMarkup(
      <VerticalProgress value="50" hasSuccessBar>
        <VerticalStep done>Step 1</VerticalStep>
        <VerticalStep active>Step 2</VerticalStep>
        <VerticalStep>Step 2</VerticalStep>
      </VerticalProgress>
    ));

  it('renders a vertical step with green success check', () =>
    matchesMarkup(
      <VerticalStep done hasSuccessMarker>
        Step 1
      </VerticalStep>
    ));
});
