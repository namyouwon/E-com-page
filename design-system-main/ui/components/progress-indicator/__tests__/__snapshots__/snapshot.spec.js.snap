// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Progress Indicator renders a base progress indicator 1`] = `
<div
  className="slds-progress"
>
  <ol
    className="slds-progress__list"
  >
    <li
      className="slds-progress__item slds-is-active"
    >
      <button
        className="slds-button slds-progress__marker"
      >
        <span
          className="slds-assistive-text"
        >
          Step 1
           
          - Active
        </span>
      </button>
    </li>
    <li
      className="slds-progress__item"
    >
      <button
        className="slds-button slds-progress__marker"
      >
        <span
          className="slds-assistive-text"
        >
          Step 2
           
        </span>
      </button>
    </li>
  </ol>
  <div
    aria-label="{{Placeholder for description of progress bar}}"
    aria-valuemax="100"
    aria-valuemin="0"
    className="slds-progress-bar slds-progress-bar_x-small"
    role="progressbar"
  >
    <span
      className="slds-progress-bar__value"
      style={
        Object {
          "width": "undefined%",
        }
      }
    >
      <span
        className="slds-assistive-text"
      >
        Progress: 
        undefined%
      </span>
    </span>
  </div>
</div>
`;

exports[`Base Progress Indicator renders a base step 1`] = `
<li
  className="slds-progress__item"
>
  <button
    className="slds-button slds-progress__marker"
  >
    <span
      className="slds-assistive-text"
    >
      Step 1
       
    </span>
  </button>
</li>
`;

exports[`Base Progress Indicator renders a done base step 1`] = `
<li
  className="slds-progress__item slds-is-completed"
>
  <button
    className="slds-button slds-button_icon slds-progress__marker slds-progress__marker_icon"
    onClick={[Function]}
    title="Step 1 - Completed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#success"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Step 1 - Completed
    </span>
  </button>
</li>
`;

exports[`Base Progress Indicator renders an active base step 1`] = `
<li
  className="slds-progress__item slds-is-active"
>
  <button
    className="slds-button slds-progress__marker"
  >
    <span
      className="slds-assistive-text"
    >
      Step 1
       
      - Active
    </span>
  </button>
</li>
`;

exports[`Base Progress Indicator renders an active base step with aria-describedby 1`] = `
<li
  className="slds-progress__item slds-is-active"
>
  <button
    aria-describedby="step-1-tooltip"
    className="slds-button slds-progress__marker"
  >
    <span
      className="slds-assistive-text"
    >
      Step 1
       
      - Active
    </span>
  </button>
</li>
`;

exports[`Base Progress Indicator renders an error base step 1`] = `
<li
  className="slds-progress__item slds-has-error"
>
  <button
    className="slds-button slds-button_icon slds-progress__marker slds-progress__marker_icon"
    onClick={[Function]}
    title="Step 1 - Error"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#error"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Step 1 - Error
    </span>
  </button>
</li>
`;

exports[`Vertical Progress Indicator renders a done vertical step 1`] = `
<li
  className="slds-progress__item slds-is-completed"
>
  <span
    className="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon"
    title="Complete"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-icon_xx-small"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#success"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Complete
    </span>
  </span>
  <div
    className="slds-progress__item_content slds-grid slds-grid_align-spread"
  >
    Step 1
  </div>
</li>
`;

exports[`Vertical Progress Indicator renders a vertical progress indicator 1`] = `
<div
  className="slds-progress slds-progress_vertical"
>
  <ol
    className="slds-progress__list"
  >
    <li
      className="slds-progress__item slds-is-active"
    >
      <div
        className="slds-progress__marker"
      >
        <span
          className="slds-assistive-text"
        >
          Active
        </span>
      </div>
      <div
        className="slds-progress__item_content slds-grid slds-grid_align-spread"
      >
        Step 1
      </div>
    </li>
    <li
      className="slds-progress__item"
    >
      <div
        className="slds-progress__marker"
      />
      <div
        className="slds-progress__item_content slds-grid slds-grid_align-spread"
      >
        Step 2
      </div>
    </li>
  </ol>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={0}
    role="progressbar"
  >
    <span
      className="slds-assistive-text"
    >
      Progress: 
      0%
    </span>
  </div>
</div>
`;

exports[`Vertical Progress Indicator renders a vertical progress indicator with borders 1`] = `
<div
  className="slds-progress slds-progress_vertical"
>
  <ol
    className="slds-progress__list slds-progress__list-bordered"
  >
    <li
      className="slds-progress__item slds-is-completed"
    >
      <span
        className="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon"
        title="Complete"
      >
        <svg
          aria-hidden={true}
          className="slds-icon slds-icon_xx-small"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#success"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Complete
        </span>
      </span>
      <div
        className="slds-progress__item_content slds-grid slds-grid_align-spread"
      >
        Step 1
      </div>
    </li>
    <li
      className="slds-progress__item slds-is-active"
    >
      <div
        className="slds-progress__marker"
      >
        <span
          className="slds-assistive-text"
        >
          Active
        </span>
      </div>
      <div
        className="slds-progress__item_content slds-grid slds-grid_align-spread"
      >
        Step 2
      </div>
    </li>
    <li
      className="slds-progress__item"
    >
      <div
        className="slds-progress__marker"
      />
      <div
        className="slds-progress__item_content slds-grid slds-grid_align-spread"
      >
        Step 2
      </div>
    </li>
  </ol>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
    role="progressbar"
  >
    <span
      className="slds-assistive-text"
    >
      Progress: 
      50%
    </span>
  </div>
</div>
`;

exports[`Vertical Progress Indicator renders a vertical progress indicator with green bar 1`] = `
<div
  className="slds-progress slds-progress_vertical slds-progress_success"
>
  <ol
    className="slds-progress__list"
  >
    <li
      className="slds-progress__item slds-is-completed"
    >
      <span
        className="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon"
        title="Complete"
      >
        <svg
          aria-hidden={true}
          className="slds-icon slds-icon_xx-small"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#success"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Complete
        </span>
      </span>
      <div
        className="slds-progress__item_content slds-grid slds-grid_align-spread"
      >
        Step 1
      </div>
    </li>
    <li
      className="slds-progress__item slds-is-active"
    >
      <div
        className="slds-progress__marker"
      >
        <span
          className="slds-assistive-text"
        >
          Active
        </span>
      </div>
      <div
        className="slds-progress__item_content slds-grid slds-grid_align-spread"
      >
        Step 2
      </div>
    </li>
    <li
      className="slds-progress__item"
    >
      <div
        className="slds-progress__marker"
      />
      <div
        className="slds-progress__item_content slds-grid slds-grid_align-spread"
      >
        Step 2
      </div>
    </li>
  </ol>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
    role="progressbar"
  >
    <span
      className="slds-assistive-text"
    >
      Progress: 
      50%
    </span>
  </div>
</div>
`;

exports[`Vertical Progress Indicator renders a vertical progress indicator with some progress 1`] = `
<div
  className="slds-progress slds-progress_vertical"
>
  <ol
    className="slds-progress__list"
  >
    <li
      className="slds-progress__item slds-is-completed"
    >
      <span
        className="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon"
        title="Complete"
      >
        <svg
          aria-hidden={true}
          className="slds-icon slds-icon_xx-small"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#success"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Complete
        </span>
      </span>
      <div
        className="slds-progress__item_content slds-grid slds-grid_align-spread"
      >
        Step 1
      </div>
    </li>
    <li
      className="slds-progress__item slds-is-active"
    >
      <div
        className="slds-progress__marker"
      >
        <span
          className="slds-assistive-text"
        >
          Active
        </span>
      </div>
      <div
        className="slds-progress__item_content slds-grid slds-grid_align-spread"
      >
        Step 2
      </div>
    </li>
    <li
      className="slds-progress__item"
    >
      <div
        className="slds-progress__marker"
      />
      <div
        className="slds-progress__item_content slds-grid slds-grid_align-spread"
      >
        Step 2
      </div>
    </li>
  </ol>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
    role="progressbar"
  >
    <span
      className="slds-assistive-text"
    >
      Progress: 
      50%
    </span>
  </div>
</div>
`;

exports[`Vertical Progress Indicator renders a vertical step 1`] = `
<li
  className="slds-progress__item"
>
  <div
    className="slds-progress__marker"
  />
  <div
    className="slds-progress__item_content slds-grid slds-grid_align-spread"
  >
    Step 1
  </div>
</li>
`;

exports[`Vertical Progress Indicator renders a vertical step with green success check 1`] = `
<li
  className="slds-progress__item slds-is-completed"
>
  <span
    className="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-progress__marker_icon-success"
    title="Complete"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-icon_xx-small"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#success"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Complete
    </span>
  </span>
  <div
    className="slds-progress__item_content slds-grid slds-grid_align-spread"
  >
    Step 1
  </div>
</li>
`;

exports[`Vertical Progress Indicator renders an active vertical step 1`] = `
<li
  className="slds-progress__item slds-is-active"
>
  <div
    className="slds-progress__marker"
  >
    <span
      className="slds-assistive-text"
    >
      Active
    </span>
  </div>
  <div
    className="slds-progress__item_content slds-grid slds-grid_align-spread"
  >
    Step 1
  </div>
</li>
`;

exports[`Vertical Progress Indicator renders an error vertical step 1`] = `
<li
  className="slds-progress__item slds-has-error"
>
  <span
    className="slds-icon_container slds-icon-utility-error slds-progress__marker slds-progress__marker_icon"
    title="Error"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-icon_xx-small"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#error"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Error
    </span>
  </span>
  <div
    className="slds-progress__item_content slds-grid slds-grid_align-spread"
  >
    Step 1
  </div>
</li>
`;
